rotkehlchen/balances/historical.py

```diff
@@ -35,16 +53,21 @@ def get_balances(self, timestamp: Timestamp) -> dict[Asset, HistoricalBalance]:
 
         May raise:
         - NotFoundError if balance for the given timestamp does not exist.
+        - DeserializationError if there is a problem deserializing an event or trade from DB.
         """
         events, main_currency = self._get_events_and_currency(to_ts=timestamp)
         if len(events) == 0:
             raise NotFoundError('No historical data found until the given timestamp.')
 
-        amounts = self._calculate_balance_from_events(events)
+        current_balances: dict[Asset, FVal] = defaultdict(FVal)
+        for event in events:
+            if self._update_balances(event, current_balances) is not None:
```

Comment:
you did not use kwargs here and marked [the comment](https://github.com/rotki/rotki/pull/9288#discussion_r1929153243) as resolved. Please do not do this.